src/interpreter/compiler.ts

```diff
@@ -0,0 +1,290 @@
+      compile(item, state);
+    });
+  },
+  BreakStatement: (node: BreakStatement, state: CompilerState) => {},
+  CallExpression: (node: CallExpression, state: CompilerState) => {
+    if (!isIdentifier(node.callee)) {
+      throw new InvalidCallError('Cannot call non-identifier.');
+    }
+    const functionEntry = getFunctionSymbolTableEntry(
+      node.callee.name,
+      state.symbolTable
+    );
+    if (functionEntry.numOfParams !== node.arguments.length) {
+      throw new InvalidCallError(
+        `Function takes in ${functionEntry.numOfParams} arguments but ${node.arguments.length} arguments were passed in.`
+      );
+    }
+
+    compile(node.callee, state);
+    // Compile in reverse order so that last argument is lower in the stash,
+    // and the first argument is higher.
+    for (let i = node.arguments.length - 1; i >= 0; i--) {
+      compile(node.arguments[i], state);
+    }
+    const callInstr = constructCallInstr(
+      node.arguments.length,
+      functionEntry.numOfVariables
+    );
+    state.instructions.push(callInstr);
+  },
+  ConditionalExpression: (
+    node: ConditionalExpression,
+    state: CompilerState
+  ) => {
+    compile(node.predicate, state);
+    const jumpOnFalseInstr = constructJumpOnFalseInstr(PLACEHOLDER_ADDRESS);
+    state.instructions.push(jumpOnFalseInstr);
+    compile(node.consequent, state);
+    const gotoInstr = constructGotoInstr(PLACEHOLDER_ADDRESS);
+    state.instructions.push(gotoInstr);
+    jumpOnFalseInstr.instrAddress = state.instructions.length;
+    compile(node.alternate, state);
+    gotoInstr.instrAddress = state.instructions.length;
+  },
+  Constant: (node: Constant, state: CompilerState) => {
+    const loadConstantInstr = constructLoadConstantInstr(node.value);
+    state.instructions.push(loadConstantInstr);
+  },
+  ContinueStatement: (node: ContinueStatement, state: CompilerState) => {},
+  DefaultStatement: (node: DefaultStatement, state: CompilerState) => {},
+  DoWhileStatement: (node: DoWhileStatement, state: CompilerState) => {},
+  EmptyStatement: (node: EmptyStatement, state: CompilerState) => {},
+  ExpressionStatement: (node: ExpressionStatement, state: CompilerState) => {},
+  ForStatement: (node: ForStatement, state: CompilerState) => {},
+  FunctionDeclaration: (node: FunctionDeclaration, state: CompilerState) => {
+    const loadFunctionInstr = constructLoadFunctionInstr(PLACEHOLDER_ADDRESS);
+    state.instructions.push(loadFunctionInstr);
+    const gotoInstr = constructGotoInstr(PLACEHOLDER_ADDRESS);
+    state.instructions.push(gotoInstr);
+
+    loadFunctionInstr.functionInstrAddress = state.instructions.length;
+
+    // TODO: Replace param declarations when parameter list is supported
+    const functionSymbolTable = addFunctionSymbolTableEntries(
+      [],
+      node,
+      state.symbolTable
+    );
+    if (!isEmptyStatement(node.body)) {
+      node.body.items.forEach((item) => {
+        state.symbolTable = functionSymbolTable;
```

Comment:
Similar question as above. It might be good to define somewhere (maybe on the type?) what `symbolTable` in `CompilerState` should refer to at any point of time when compiling.